Fix logic for detection of Docker environments in a multi-container enviroment

styrmis opened the pull request:
This includes the fix from https://github.com/tawan/active-elastic-job/pull/107 for the detection of whether the app runs in a Docker container or not (to support execution within a multi-container Elastic Beanstalk environment) and also a relaxing of the match for Docker host IPs, again to support multi-container environments.

## References

- https://github.com/tawan/active-elastic-job/pull/107
- https://github.com/tawan/active-elastic-job/issues/94
- https://github.com/tawan/active-elastic-job/issues/104

joeyparis commented:
It looks like I need additional access to merge the V3 branch (with fixed Travis testing) into this PR. Can you allow me to do that, or merge branch V3 in yourself? Thank you.

styrmis commented:
Ok, I've updated this branch against the V3 branch. I was unable to merge V3
directly so I rebased this branch against V3, hope that's ok.

Thank you for reviewing it!

joeyparis commented:
Shouldn't be a problem! At this point, I'm more interested in getting to a V3 alpha with all the outstanding pull-requests merged in. Thanks for coming back to this after so much time.
